### PR TITLE
EMSUSD-1692 - Adds LookdevXUsd Extension

### PR DESCRIPTION
**Pull request by AramAzhari-adsk**
This change adds USD support for LookdevX to allow compiling with different versions of USD.

### CODE REVIEW

File: plugin/adsk/plugin/CMakeLists.txt [hunk 1]
```diff
@@ -100,6 +100,12 @@ if(IS_MACOSX OR IS_LINUX)
     )
 endif()
 
+# For initializing the handlers of lookdevXUsd
+target_link_libraries(${TARGET_NAME}
+    PRIVATE
+        lookdevXUsd
+)
```
Comment (seando-adsk) [inline plugin/adsk/plugin/CMakeLists.txt:107]: This should be wrapped in the cmake flag. We only need to link with it when its built.

Comment (seando-adsk) [reply plugin/adsk/plugin/CMakeLists.txt:107]: And you'll need to add a `target_compile_definitions` so you can use ifdef in the plugin.cpp below.

File: plugin/adsk/plugin/plugin.cpp [hunk 2]
```diff
@@ -56,6 +56,7 @@
 #include <ufe/runTimeMgr.h>
 
 #include <basePxrUsdPreviewSurface/usdPreviewSurfacePlugin.h>
+#include <lookdevXUsd/LookdevXUsd.h>
```
Comment (seando-adsk) [inline plugin/adsk/plugin/plugin.cpp:59]: This line and the two calls below need to be wrapped in `#ifdef` and only compiled in when we built LookdevXUsd.

**Review by seando-adsk — CHANGES_REQUESTED** [review]
A couple of minor changes are required.

**Comment by AramAzhari-adsk** [comment]
Currently waiting on https://github.com/Autodesk/maya-usd/pull/4042 to be integrated into the system that the clang formatting will be check against.

File: test/lib/ufe/CMakeLists.txt [hunk 3]
```diff
@@ -173,6 +173,7 @@ foreach(script ${TEST_SCRIPT_FILES})
             "UFE_SCENE_SEGMENT_HANDLER_ROOT_PATH=${UFE_SCENE_SEGMENT_HANDLER_ROOT_PATH}"
             "UFE_VOLUME_LIGHTS_SUPPORT=${UFE_VOLUME_LIGHTS_SUPPORT}"
             "UFE_SCENEITEM_HAS_METADATA=${UFE_SCENEITEM_HAS_METADATA}"
+            "BUILD_LOOKDEVXUSD_LIBRARY=${BUILD_LOOKDEVXUSD_LIBRARY}"
```
[the remark this answers is not in the document]
Comment (seando-adsk) [reply test/lib/ufe/CMakeLists.txt]: Ah yes, sorry got them reversed.

**Review by frohnej-adsk — DISMISSED** [review]
LGTM

**Review by seando-adsk — APPROVED** [review]
Excellent work thanks

**Pull request merged**